File: internal/httpapi/accounts.go
package httpapi

import (
	"net/http"
	"xuetangx/internal/domain"
)

func accountView(a domain.Account) map[string]any {
	var user any
	if a.ID != "" {
		user = map[string]any{"user_id": a.UserID, "id": a.UserID, "name": a.Name, "username": a.Name}
	}
	return map[string]any{"connected": a.Connected, "role": a.Role, "userId": a.UserID, "user": user, "connectedAt": a.ConnectedAt}
}
func (s *Server) accountRoutes(m *http.ServeMux) {
	m.Handle("GET /api/session", s.require(func(w http.ResponseWriter, r *http.Request) {
		a, e := s.Accounts.Get(r.Context(), principal(r).User.ID, "primary")
		if e != nil {
			writeError(w, e)
			return
		}
		writeJSON(w, 200, accountView(a))
	}))
	for _, role := range []string{"primary", "test"} {
		connect, disconnect := "/api/cookie", "/api/disconnect"
		if role == "test" {
			connect, disconnect = "/api/test-cookie", "/api/test-disconnect"
		}
		m.Handle("POST "+connect, s.require(func(w http.ResponseWriter, r *http.Request) {
			var v struct{ Cookie string }
			if e := body(w, r, &v); e != nil {
				writeError(w, e)
				return
			}
			a, e := s.Accounts.Connect(r.Context(), principal(r).User.ID, role, v.Cookie)
			if e != nil {
				writeError(w, e)
				return
			}
			out := accountView(a)
			out["ok"] = true
			writeJSON(w, 200, out)
		}))
		m.Handle("POST "+disconnect, s.require(func(w http.ResponseWriter, r *http.Request) {
			if e := s.Accounts.Disconnect(r.Context(), principal(r).User.ID, role); e != nil {
				writeError(w, e)
				return
			}
			writeJSON(w, 200, map[string]bool{"ok": true})
		}))
	}
}

File: server/workflow/routes/accounts.js
function register(app, { runtime, handle, primary, courseList, legacyState, latest, invalidateCourses }) {
  app.get('/api/session', (req, res) => res.json(runtime.accounts.summary('primary')));
  app.post('/api/cookie', handle(async (req, res) => { const connected = await runtime.accounts.connect('primary', req.body?.cookie); invalidateCourses(); res.json({ ok: true, user: connected.user }); }));
  app.post('/api/disconnect', (req, res) => { runtime.accounts.clear('primary'); res.json({ ok: true }); });
  app.post('/api/test-cookie', handle(async (req, res) => res.json({ ok: true, ...await runtime.accounts.connect('test', req.body?.cookie) })));
  app.post('/api/test-disconnect', (req, res) => { runtime.accounts.clear('test'); res.json({ ok: true }); });
}
module.exports = { register };
